functests/1_performance/cpu_management.go

```diff
@@ -153,6 +157,121 @@ var _ = Describe("[rfe_id:27363][performance] CPU Management", func() {
 			table.Entry("Guaranteed POD should work on isolated cpu", true),
 		)
 	})
+
+	When("pod runs with the CPU load balancing runtime class", func() {
+		var testpod *corev1.Pod
+		var defaultFlags []int
+
+		getCPUSchedulingDomainFlags := func(cpu int) ([]int, error) {
+			cmd := []string{"/bin/bash", "-c", fmt.Sprintf("cat /proc/sys/kernel/sched_domain/cpu%d/domain*/flags", cpu)}
+			out, err := nodes.ExecCommandOnNode(cmd, workerRTNode)
+			if err != nil {
+				return nil, err
+			}
+
+			var domainsFlags []int
+			for _, domainFlags := range strings.Split(out, "\n") {
+				flags, err := strconv.Atoi(domainFlags)
+				if err != nil {
+					return nil, err
+				}
+				domainsFlags = append(domainsFlags, flags)
+			}
+
+			sort.Ints(domainsFlags)
+			return domainsFlags, nil
+		}
+
+		BeforeEach(func() {
+			// get the default value for sched_domain flags, we will check the flags only for the CPU 0, because on the
+			// clean system it should be the same for all CPUs
+			var err error
+			defaultFlags, err = getCPUSchedulingDomainFlags(0)
+			Expect(err).ToNot(HaveOccurred())
+
+			testpod = pods.GetTestPod()
+			testpod.Namespace = testutils.NamespaceTesting
+
+			cpus := resource.MustParse("1")
+			memory := resource.MustParse("256Mi")
+
+			// change pod resource requirements, to change the pod QoS class to guaranteed
+			testpod.Spec.Containers[0].Resources = corev1.ResourceRequirements{
+				Limits: corev1.ResourceList{
+					corev1.ResourceCPU:    cpus,
+					corev1.ResourceMemory: memory,
+				},
+			}
+
+			// use the CPU load balancing runtime class
+			runtimeClassName := components.GetComponentName(profile.Name, components.ComponentNamePrefix)
+			testpod.Spec.RuntimeClassName = &runtimeClassName
+		})
+
+		AfterEach(func() {
+			// it possible that the VM already was deleted as part of the test, in this case we want to skip teardown
+			key := types.NamespacedName{
+				Name:      testpod.Name,
+				Namespace: testpod.Namespace,
+			}
+			err := testclient.Client.Get(context.TODO(), key, testpod)
+			if errors.IsNotFound(err) {
+				return
+			}
+
+			err = testclient.Client.Delete(context.TODO(), testpod)
+			Expect(err).ToNot(HaveOccurred())
+
+			err = pods.WaitForDeletion(testpod, 60*time.Second)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("should disable CPU load balancing for CPU's used by the pod", func() {
+			By("Starting the pod")
+			err := testclient.Client.Create(context.TODO(), testpod)
+			Expect(err).ToNot(HaveOccurred())
+
+			err = pods.WaitForCondition(testpod, corev1.PodReady, corev1.ConditionTrue, 10*time.Minute)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Checking what CPU the pod is using")
+			cmd := []string{"/bin/bash", "-c", "ps -o psr $(pgrep -n test) | tail -1"}
```

Comment:
In the future we probably should make the test pod just record the pid of the main process somewhere, so tests like this are as trivial (and more robust and future proof) as to read the pid from a pidfile.

Reply:
This check does not verify correctly the CPU pinning(because we do not have test process, only the `/usr/libexec/crio/conmon` processes), I will check the cpu pinning under the container cgroup.

Reply:
ok, then I need to re-read more carefully the code :)